### PR TITLE
Warn instead of failing when renaming metrics using metric_relabel_configs

### LINKED ISSUE #5001: Renaming metrics with prometheus receiver produces unknown-type metrics
**Describe the bug**
If the `__name__` label is changed using prometheus relabel configs, the prometheus receiver fails to populate metric metadata.

The root cause is that the prometheus receiver [looks up metric metadata](https://github.com/open-telemetry/opentelemetry-collector/blob/575600b8b63e448c7d1e6559b90b971b8fd58949/receiver/prometheusreceiver/internal/metricfamily.go#L53) using the _final_ metric name.  However, the prometheus server [updates the metadata cache](https://github.com/prometheus/prometheus/blob/e5478983c43cb839a77c144f953b9ea25783b995/scrape/scrape.go#L1280) before [applying relabel rules](https://github.com/prometheus/prometheus/blob/e5478983c43cb839a77c144f953b9ea25783b995/scrape/scrape.go#L1326), meaning the metadata cache stores metadata based on the _initial_ metric name.  The metadata lookup fails, and it is left empty.

**Steps to reproduce**
Run a collector with the config below.

The collector produces the logs:
```
2020-12-16T21:04:44.840Z	info	internal/metrics_adjuster.go:357	Adjust - skipping unexpected point	{"component_kind": "receiver", "component_type": "prometheus", "component_name": "prometheus", "type": "UNSPECIFIED"}
2020-12-16T21:04:45.035Z	INFO	loggingexporter/logging_exporter.go:361	MetricsExporter	{"#metrics": 1}
2020-12-16T21:04:45.035Z	DEBUG	loggingexporter/logging_exporter.go:388	ResourceMetrics #0
Resource labels:
     -> service.name: STRING(prometheusreceiver)
     -> host.name: STRING(localhost)
     -> port: STRING(8888)
     -> scheme: STRING(http)
InstrumentationLibraryMetrics #0
InstrumentationLibrary
Metric #0
Descriptor:
     -> Name:
     -> Description:
     -> Unit:
     -> DataType: None
```

Note the empty Name, Description, Unit, and DataType.

**What did you expect to see?**
I expected metrics to be renamed, and otherwise emitted normally.

**What did you see instead?**
Metrics are dropped during [conversion from OpenCensus format to OpenTelemetry format](https://github.com/open-telemetry/opentelemetry-collector/blob/575600b8b63e448c7d1e6559b90b971b8fd58949/translator/internaldata/oc_to_metrics.go#L179) because the metric descriptor type is Unknown.

**What version did you use?**
Version: otel/opentelemetry-collector-contrib-dev@sha256:0dbc61590cc04678997173fb378c875e2733ff2e443d75a7957a340d4b2bb9ef (latest)

**What config did you use?**
Config: (e.g. the yaml config file)

```yaml
receivers:
  prometheus:
    config:
      global:
        scrape_interval: 10s
      scrape_configs:
      - job_name: 'prometheusreceiver'
        static_configs:
        - targets: [localhost:8888]
        metric_relabel_configs:
        # filter out all metrics except otelcol_process_cpu_seconds so it is easier to read logs
        - source_labels: [ __name__ ]
          regex: "otelcol_process_cpu_seconds"
          action: keep
        # rename otelcol_process_cpu_seconds to process_cpu_seconds by replacing the __name__ label
        - source_labels: [ __name__ ]
          regex: "otelcol_(.*)"
          action: replace
          target_label: __name__
exporters:
  logging:
    logLevel: debug
service:
  pipelines:
    metrics:
      receivers: [prometheus]
      exporters: [logging]
```

**Additional context**
I was able to determine the root cause above by adding additional debug statements and rebuilding the collector.

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5001

When that issue was filed, we dropped unknown-typed metrics.  Now that we convert them to gauges, users may want to use metric_relabel_configs to rename metrics, even if they lose metadata.  In the future, we have other enhancements planned to better-support unknown-typed metrics.

Change the error to a warning.  It is still likely not a best practice to use relabel configs for this purpose, but for users copy-pasting prometheus configs, it is best not to fail if we can help it.

### CODE REVIEW

File: receiver/prometheusreceiver/config_test.go [hunk 1]
```diff
@@ -119,8 +121,8 @@ func TestLoadConfigFailsOnUnknownPrometheusSection(t *testing.T) {
 	require.Error(t, component.UnmarshalConfig(sub, cfg))
 }
 
-// Renaming is not allowed
-func TestLoadConfigFailsOnRenameDisallowed(t *testing.T) {
+// Renaming emits a warning
+func TestConfigWarningsOnRenameDisallowed(t *testing.T) {
 	cm, err := confmaptest.LoadConf(filepath.Join("testdata", "invalid-config-prometheus-relabel.yaml"))
```
Comment (crobert-1) [inline receiver/prometheusreceiver/config_test.go:126]: Nit: the name `invalid-config` is no longer correct, it'd be good to update along with this change.

**Review by crobert-1 — APPROVED** [review]

File: receiver/prometheusreceiver/config_test.go [hunk 2]
```diff
@@ -119,8 +121,8 @@ func TestLoadConfigFailsOnUnknownPrometheusSection(t *testing.T) {
 	require.Error(t, component.UnmarshalConfig(sub, cfg))
 }
 
-// Renaming is not allowed
-func TestLoadConfigFailsOnRenameDisallowed(t *testing.T) {
+// Renaming emits a warning
+func TestConfigWarningsOnRenameDisallowed(t *testing.T) {
```
Comment (crobert-1) [inline receiver/prometheusreceiver/config_test.go:125]: Would it make sense to add a test for this that goes through the main `createMetricsReceiver` to make sure we're logging once for each rename, and still hitting the method properly? My concern is just making sure it works as a whole, not just at the direct call/unit test level.

File: receiver/prometheusreceiver/config_test.go (hunk 1 above)
Comment (dashpole) [reply receiver/prometheusreceiver/config_test.go]: changed to warning-config

**Review by crobert-1 — APPROVED** [review]

**Comment by decimalst** [comment]
Hey folks, any traction on this? I was hoping to use this in the context of a prometheus receiver not exporting to prometheus

**Review by songy23 — APPROVED** [review]
LGTM

**Pull request merged**